base/dates/io-util.jl

```diff
@@ -129,12 +118,11 @@ end
 end
 
 @inline function tryparsenext_word(str, i, len, locale, maxchars=0)
-    j = 1
-    while (maxchars <= 0 || j <= maxchars) && i <= len
+    max_pos = maxchars <= 0 ? len : min(chr2ind(str, ind2chr(str,i) + maxchars - 1), len)
+    while i <= max_pos
         c, ii = next(str, i)
```

Comment:
you can actually also put `@inbounds` on this line and pretty much everywhere `next` is called to squeeze out a few more nanoseconds!